release/ray_release/file_manager/job_file_manager.py

```diff
@@ -132,24 +120,30 @@ def upload(self, source: Optional[str] = None, target: Optional[str] = None):
         assert isinstance(target, str)
 
         remote_upload_to = self._generate_tmp_s3_path()
+
         # local source -> s3
-        self.s3_client.upload_file(
-            Filename=source,
-            Bucket=self.bucket,
-            Key=remote_upload_to,
-        )
+        def upload():
+            self.s3_client.upload_file(
+                Filename=source,
+                Bucket=self.bucket,
+                Key=remote_upload_to,
+            )
+
+        self._run_with_retry(upload)
+
         # s3 -> remote target
         bucket_address = f"s3://{self.bucket}/{remote_upload_to}"
-        cid = self._run_job(
+        retcode, _ = self.job_manager.run_and_wait(
             "pip install -q awscli && " f"aws s3 cp {bucket_address} {target}",
             {},
         )
-        retcode, duration = self._wait_job(cid)
 
         if retcode != 0:
             raise FileUploadError(f"Error uploading file {source} to {target}")
 
         try:
-            self.s3_client.delete_object(Bucket=self.bucket, Key=remote_upload_to)
+            self._run_with_retry(
+                self.s3_client.delete_object(Bucket=self.bucket, Key=remote_upload_to)
+            )
```

Comment:
This has to be a `lambda`, right?

By the way, let's make this the same in all three locations (lines 72, 106, 146) - right now we have a `def`, a `lambda` and the raw function (which is probably wrong). Let's go with a lambda everywhere instead

Reply:
Oh yeah. Fixed all